pydantic/networks.py

```diff
@@ -109,6 +110,8 @@ class AnyUrl(str):
     allowed_schemes: Optional[Set[str]] = None
     tld_required: bool = False
     user_required: bool = False
+    default_parts: Optional[Set[Tuple[str, str]]] = None
+    display_port: bool = True
```

Comment:
Hmmm I think this issue doesn't concern only the port. The question is "when a value is set by magic, should the string be changed". As a matter of fact I think you don't need this. What matters is `rebuild`
```py
host, tld, host_type, rebuild = cls.validate_host(parts)
```
which, when set to `True`, will rebuild the url

Reply:
I didn't find a good way to implement this..

Reply:
I had to reverted the "default http port" behaviour